### PR TITLE
New functions for Consent evaluation

### PR DESCRIPTION
**Pull request by michaelabuckley**

### CODE REVIEW

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnum.java [hunk 1]
```diff
@@ -39,4 +41,66 @@ public enum ConsentOperationStatusEnum {
...
 	 */
 	AUTHORIZED,
+	;
+
+	/**
+	 * Assigns ordinals to the verdicts by strength:
+	 * REJECT > AUTHORIZED > PROCEED.
+	 * @return 2/1/0 for REJECT/AUTHORIZED/PROCEED
+	 */
+	int getStrength() {
```
Comment (tadgh) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnum.java:51]: nitpick: getStrength -> getPrecedence?

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnum.java [hunk 2]
```diff
@@ -39,4 +41,66 @@ public enum ConsentOperationStatusEnum {
...
+		} else {
+			return theNextVerdict;
+		}
+	}
+
+	/**
+	 * Evaluate all verdicts together, allowing any to veto (i.e. REJECT) the operation.
+	 *
+	 * @return REJECT if any reject, AUTHORIZED if no REJECT and some AUTHORIZED, PROCEED if empty or all PROCEED
+	 */
```
Comment (tadgh) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnum.java:89]: nitpick: nice! Some example inputs -> outputs in the javadoc could be useful, e.g. 

```
[AUTHORIZED,AUTHORIZED,PROCEED,REJECT] -> REJECT
```

File: hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnumTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+class ConsentOperationStatusEnumTest {
+
+	/**
+	 * With "serial" evaluation, the first non-PROCEED verdict wins.
+	 */
+	@ParameterizedTest
+	@CsvSource(textBlock = """
+		REJECT     REJECT     REJECT    , REJECT
```
Comment (tadgh) [inline hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnumTest.java:22]: nice test

**Review by tadgh — APPROVED** [review]
Will this new behaviour bubble up into a public difference in behaviour? if so, a changelog is worthwhile.

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnum.java (hunk 1 above)
Comment (michaelabuckley) [reply hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnum.java:51]: done.

**Pull request merged**